docs/source/download.rst

```diff
@@ -110,6 +110,10 @@ examples on the :doc:`documentation page <documentation>`.
 Git instructions for developers
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+Cython >= 0.17 is required to build Firedrake. Install it using pip ::
+
+ pip install "Cython>=0.17"
```

Comment:
I think so.  Maybe also update the cython requirement in PyOP2 requirements-minimal.txt?